app/models/journable/historic_active_record_relation.rb

```
@@ -0,0 +1,290 @@
+#   Prefer `kind_of?` over `is_a?` because it reads well before vowel and consonant sounds.
+#   E.g.: `relation.kind_of? ActiveRecord::Relation`
+
+# rubocop:disable Metrics/AbcSize
+# rubocop:disable Metrics/PerceivedComplexity
+
+# In the context of the baseline-comparison feature, this class represents an active-record relation
+# that queries historic data, i.e. performs its query e.g. on the `work_package_journals` table
+# rather than the `work_packages` table.
+#
+# Usage:
+#
+#     timestamp = 1.year.ago
+#     active_record_relation = WorkPackage.where(subject: "Foo")
+#     historic_relation = Journable::HistoricActiveRecordRelation.new(active_record_relation, timestamp:)
+#
+# See also:
+#
+# - https://github.com/opf/openproject/pull/11243
+# - https://community.openproject.org/projects/openproject/work_packages/26448
+#
+class Journable::HistoricActiveRecordRelation < ActiveRecord::Relation
+  attr_accessor :timestamp
+
+  include ActiveRecord::Delegation::ClassSpecificRelation
+
+  def initialize(relation, timestamp:)
+    raise ArgumentError, "Expected ActiveRecord::Relation" unless relation.kind_of? ActiveRecord::Relation
+
+    super(relation.klass)
+    relation.instance_variables.each do |key|
+      instance_variable_set key, relation.instance_variable_get(key)
+    end
+
+    self.timestamp = timestamp
+    readonly!
+    instance_variable_set :@table, model.journal_class.arel_table
+  end
+
+  # We need to patch the `pluck` method of an active-record relation that
+  # queries historic data (i.e. journal data). Otherwise, `pluck(:id)`
+  # would return the `id` of the journal table rather than the `id` of the
+  # journable table, which would be expected from the syntax:
+  #
+  #     WorkPackage.where(assigned_to_id: 123).at_timestamp(1.year.ago).pluck(:id)
+  #
+  def pluck(*column_names)
+    column_names.map! { |column_name| column_name == :id ? 'journals.journable_id' : column_name }
+    arel
+    super
+  end
```

There are more properties for which the behaviour might be surprising but those will fail hard:
* created_at
* updated_at
* lock_version
* position

From the top of my head, I don't think they are used within the codebase so it might be acceptable to let them simply fail hard until they are employed. On the other hand, it is a recurring pattern to have to map the properties so if some map could be established that is used in the various cases, further errors can be avoided. At least `created_at` and `updated_at` are non work package specific.

---

👍 resolved in https://github.com/opf/openproject/pull/11243/commits/41470cac7faebde554ebe98c7a1539adffef2bc9